lib/Sema/CSSimplify.cpp

```diff
@@ -836,16 +757,35 @@ class ArgumentFailureTracker : public MatchCallArgumentListener {
     return false;
   }
 
+  bool incorrectLabel(unsigned paramIndex, unsigned impact) {
+    if (!CS.shouldAttemptFixes())
+      return true;
+
+    // TODO(diagnostics): If re-labeling is mixed with extra arguments,
+    // let's produce a fix only for extraneous arguments for now,
+    // because they'd share a locator path which (currently) means
+    // one fix would overwrite another.
+    if (!ExtraArguments.empty()) {
+      CS.increaseScore(SK_Fix);
+      return false;
+    }
+
+    auto *fix = RelabelSingleArgument::create(CS, Bindings[paramIndex].front(),
+                                              Parameters[paramIndex].getLabel(),
+                                              CS.getConstraintLocator(Locator));
```

Comment:
OK, I change approach so.

Actually I have tried to locate this fix at `CallExpr -> apply argument -> arg to param`.
It could print multiple diagnostics without using `coalesceAndDiagnose`.
But another problem happened on it.

When both type error and label error exists,

e.g.
```swift
func f(aa: Int, bb: Int) {}
f(aax: "0", bb: 1)
```

This logic detects labeling failure as other argument mismatch.

https://github.com/apple/swift/blob/f4fee7a34b4ad657d86280fb5858c4fd2bf1f276/lib/Sema/CSSimplify.cpp#L3422-L3433

And solver stopped with ambiguous expression.

I will try to avoid this problem and make them works.

Reply:
I'd suggest you to try and use a new locator element `Label` specifically for re-labeling failures.

Reply:
I thought that we could handle different kinds of fixes with the same locator but apparently that is not so.

Reply:
I will try to add new locator element.